### ISSUE TITLE
Switching to Hydra (GL) viewport crashes Maya in refactoring_sandbox branch

### ISSUE THREAD

**Issue by BigRoy** [Autodesk/maya-usd#68]
### Issue

I've just compiled `maya-usd` with `refactoring_sandbox` using [the latest commit](https://github.com/Autodesk/maya-usd/commit/d0bb624e3aba2906a9fe4a09d4972324b237a3e0) which instantly crashes Maya 2019.2 with a fatal error when switching the viewport to Hydra (GL).  This even occurs in an empty new scene.

Maya Crash log: 
[MayaCrashLog191025.1356.log](https://github.com/Autodesk/maya-usd/files/3771902/MayaCrashLog191025.1356.log)

I'm not sure where the issue was introduced but it could be with PR https://github.com/Autodesk/maya-usd/pull/59 which mentions changes to the viewport 2.0 but not explicitly to Hydra.

---

About 15 days ago I was able to compile the latest of `refactoring_sandbox` branch at the time and load up the Hydra viewport just fine of which some notes were mentioned here https://github.com/Autodesk/maya-usd/issues/51

**Comment by BigRoy** [comment]
I also noticed this in the Output Window:
```
#############################################################################
#  VP2_RENDER_DELEGATE_PROXY is overridden to 'true'.  Default is 'false'.  #
#############################################################################
###############################################################################
Begin attempted read of a fragment graph XML file.
Attempting to read fragment XML code from: d:/dev/usd/maya-usd-build/install/RelWithDebInfo/plugin/pxr/maya/plugin/pxrUsdPreviewSurface/resources/usdPreviewSurface.xml
Resolved file path is: d:\dev\usd\maya-usd-build\install\RelWithDebInfo\plugin\pxr\maya\plugin\pxrUsdPreviewSurface\resources\usdPreviewSurface.xml
Attempting to read fragment XML code from a buffer of length 9192
Errors loading XML:
-------------------------------------------------------------------------------
OGS WARNING: The input parameter of a connection 'vectorInput' from fragment 'mayaShaderGeom_Float4GetX' was not found.
OGS WARNING: The input parameter of a connection 'vectorInput' from fragment 'mayaShaderGeom_Float4GetY' was not found.
OGS WARNING: The input parameter of a connection 'vectorInput' from fragment 'mayaShaderGeom_Float4GetZ' was not found.
OGS WARNING: The input parameter of a connection 'vectorInput' from fragment 'mayaShaderGeom_Float4GetW' was not found.
OGS WARNING: The output parameter of a connection 'floatOutput' from fragment 'mayaShaderGeom_Float4GetX' was not found.
OGS WARNING: The output parameter of a connection 'floatOutput' from fragment 'mayaShaderGeom_Float4GetY' was not found.
OGS WARNING: The output parameter of a connection 'floatOutput' from fragment 'mayaShaderGeom_Float4GetZ' was not found.
OGS WARNING: The output parameter of a connection 'floatOutput' from fragment 'mayaShaderGeom_Float4GetW' was not found.
-------------------------------------------------------------------------------
End attempted read of fragment XML.
###############################################################################
Stack trace:
  mtoh.mll!pxrInternal_v0_19__pxrReserved__::MtohRenderOverride::_SelectionChanged Line 694 
  mtoh.mll!pxrInternal_v0_19__pxrReserved__::MtohRenderOverride::_InitHydraResources Line 627 
  mtoh.mll!pxrInternal_v0_19__pxrReserved__::MtohRenderOverride::Render Line 462 
  mtoh.mll!pxrInternal_v0_19__pxrReserved__::HdMayaRender::execute Line 84 
  OpenMayaRender.dll!Autodesk::Maya::OpenMaya20190000::MHWRender::MRenderer::initialize
  OGSMayaBridge.dll!OGSMayaRenderer::performUserOperation
  OGSMayaBridge.dll!OGSMayaRenderer::performOverrideRenderGraph
  OGSMayaBridge.dll!OGSMayaBaseRenderer::performOverrideRender
  OGSMayaBridge.dll!OGSGeometryEvaluationOverride::~OGSGeometryEvaluationOverride
  OGSMayaBridge.dll!OGSGeometryEvaluationOverride::~OGSGeometryEvaluationOverride
  DataModel.dll!TidleRefreshCmd::refreshOGSRender
  DataModel.dll!TidleRefreshCmd::refresh
  DataModel.dll!TidleRefreshCmd::doIt
  ExtensionLayer.dll!TeventHandler::doIdles
  ExtensionLayer.dll!TidleLicenseAction::doIdleAction
  Qt5Core.dll!QObject::event
  Qt5Widgets.dll!QApplicationPrivate::notify_helper
  Qt5Widgets.dll!QApplication::notify
  ExtensionLayer.dll!TwindowManager::windowPreferencesAreEnabled
  Qt5Core.dll!QCoreApplication::notifyInternal2
  Qt5Core.dll!QEventDispatcherWin32::event
  Qt5Widgets.dll!QApplicationPrivate::notify_helper
  Qt5Widgets.dll!QApplication::notify
  ExtensionLayer.dll!TwindowManager::windowPreferencesAreEnabled
  Qt5Core.dll!QCoreApplication::notifyInternal2
  Qt5Core.dll!QCoreApplicationPrivate::sendPostedEvents
  qwindows.dll!qt_plugin_query_metadata
  Qt5Core.dll!QEventDispatcherWin32::processEvents
  USER32.dll!CallWindowProcW
  USER32.dll!DispatchMessageW
  Qt5Core.dll!QEventDispatcherWin32::processEvents
  qwindows.dll!qt_plugin_query_metadata
  Qt5Core.dll!QEventLoop::exec
  Qt5Core.dll!QCoreApplication::exec
  ExtensionLayer.dll!Tapplication::start
  maya.exe!<7FF6081C77A8>
  maya.exe!TiteratorWrapperFwd<TscenePartitions::ConstIteratorDescriptor,TiteratorWrapper<TscenePartitions::ConstIteratorDescriptor> >::operator++
  maya.exe!TiteratorWrapperFwd<TscenePartitions::ConstIteratorDescriptor,TiteratorWrapper<TscenePartitions::ConstIteratorDescriptor> >::operator++
  KERNEL32.DLL!BaseThreadInitThunk
  ntdll.dll!RtlUserThreadStart
```

And also:
![hdMaya_not_found](https://user-images.githubusercontent.com/2439881/67570200-120fe300-f731-11e9-83a8-e128248b2791.png)

This last one does seem to hint to what is going on, because indeed `hdMaya.dll` is not there.

**Comment by HdC-adsk** [comment]
Thanks for reporting the issue. I am working now for a fix.

**Comment by BigRoy** [comment]
Fixed with #79 

---

Note that [this warning](https://user-images.githubusercontent.com/2439881/67570200-120fe300-f731-11e9-83a8-e128248b2791.png) still shows, but [that is not a regression](https://github.com/Autodesk/maya-usd/pull/79#issuecomment-546440488) and thus was there before.
``` 
Coding Error: in _Load at line 248 of D:\dev\usd\USD\pxr\base\lib\plug\plugin.cpp -- Load of 'd:/dev/usd/maya-usd-build/install/RelWithDebInfo/lib/usd/hdMaya/resources/hdMaya.dll' for 'hdMaya' failed: The specified module could not be found.
```

**Comment by suekieza** [comment]
linking my issue here as well https://github.com/Autodesk/maya-usd/issues/98
